indico/modules/events/registration/client/js/components/ParticipantListAccordion.tsx

```diff
@@ -0,0 +1,263 @@
+      direction === null
+        ? [...table.rows]
+        : [...sortedRows].sort((a, b) => {
+            const textA = a.columns[columnIndex].text.toLowerCase();
+            const textB = b.columns[columnIndex].text.toLowerCase();
+
+            if (textA < textB) {
+              return direction === 'ascending' ? -1 : 1;
+            } else if (textA > textB) {
+              return direction === 'ascending' ? 1 : -1;
+            }
+
+            return 0;
+          });
+
+    setSortColumn(columnIndex);
+    setSortDirection(direction);
+    setSortedRows(sortedData);
+  };
+
+  return visibleParticipantsCount > 0 ? (
+    <Table fixed celled sortable className="table">
+      <TableHeader>
+        <TableRow className="table-row">
+          {table.show_checkin && ( // For checkin status
+            <TableHeaderCell width={1} textAlign="center">
+              <Icon name="ticket" />
+            </TableHeaderCell>
+          )}
+          {table.headers.map((headerText: string, i: number) => (
+            <TableHeaderCell
+              // eslint-disable-next-line react/no-array-index-key
+              key={i}
+              sorted={sortColumn === i ? sortDirection : undefined}
+              onClick={() => handleSort(i)}
+              title={headerText}
+            >
+              {headerText}
+            </TableHeaderCell>
+          ))}
+        </TableRow>
+      </TableHeader>
+      <TableBody>
+        {sortedRows.map((row: TableRowObj) => (
+          <TableRow key={row.id} className="table-row">
+            {table.show_checkin && (
+              <TableCell textAlign="center">
+                {row.checked_in ? (
+                  <Icon name="check" color="green" />
+                ) : (
+                  <Icon name="close" color="red" />
+                )}
+              </TableCell>
+            )}
+            {row.columns.map((col: TableColumnObj, i: number) => (
+              // eslint-disable-next-line react/no-array-index-key
+              <TableCell key={i} title={col.text}>
+                {col.is_picture && col.text ? (
+                  <img src={col.text} className="cell-img" />
+                ) : (
+                  col.text
+                )}
+              </TableCell>
+            ))}
+          </TableRow>
+        ))}
+      </TableBody>
+      {hasInvisibleParticipants && (
+        <TableFooter>
+          <TableRow>
+            <TableHeaderCell colSpan={table.headers.length}>
+              <ParticipantCountTranslationHidden count={hiddenParticipantsCount} />
+            </TableHeaderCell>
+          </TableRow>
+        </TableFooter>
+      )}
+    </Table>
+  ) : (
+    <Message>
+      {totalParticipantCount > 0 ? (
+        <ParticipantCountTranslationHidden count={hiddenParticipantsCount} />
+      ) : (
+        <Translate>No participants registered</Translate>
+      )}
+    </Message>
+  );
+}
+
+function AccordionParticipantsItem({
+  index,
+  table,
+  children = null,
```

Comment:
If `children` are never used, I'd remove them

Reply:
I did it for reusability in other contexts but we can always add that if we care for it